### ISSUE TITLE
apt/dpkg sporadic failures with --fakeroot

### ISSUE THREAD

**Issue by Zohman** [apptainer/singularity#3842]
Hi,

i wonder what can be done to allow --fakeroot when native packages
try to chown/chgrp/usradd/groupadd during build?

some failures on  Operation not permitted /  Permission denied causing the build to go fatal,
where with sudo it works..

## Version of Singularity:

3.3.0-rc.1

### Expected behavior

not to fail when native packages try chown/chgrp/usradd/groupadd

### Actual behavior

failures when installing via apt package manager for debian/ubuntu..
example:

+ apt-get -y install libsystemd0
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages will be upgraded:
  libsystemd0
1 upgraded, 0 newly installed, 0 to remove and 6 not upgraded.
Need to get 204 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libsystemd0 amd64 237-3ubuntu10.23 [204 kB]
Fetched 204 kB in 0s (0 B/s)        
debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 4658 files and directories currently installed.)
Preparing to unpack .../libsystemd0_237-3ubuntu10.23_amd64.deb ...
Unpacking libsystemd0:amd64 (237-3ubuntu10.23) over (237-3ubuntu10.22) ...
dpkg: error processing archive /var/cache/apt/archives/libsystemd0_237-3ubuntu10.23_amd64.deb (--unpack):
 unable to chown backup symlink for './lib/x86_64-linux-gnu/libsystemd.so.0': Operation not permitted
Errors were encountered while processing:
 /var/cache/apt/archives/libsystemd0_237-3ubuntu10.23_amd64.deb
W: chown to _apt:root of directory /var/cache/apt/archives/partial failed - SetupAPTPartialDirectory (1: Operation not permitted)
W: chown to _apt:root of directory /var/lib/apt/lists/auxfiles failed - SetupAPTPartialDirectory (1: Operation not permitted)
W: Download is performed unsandboxed as root as file '/var/cache/apt/archives/partial/libsystemd0_237-3ubuntu10.23_amd64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)
W: chown to root:adm of file /var/log/apt/term.log failed - OpenLog (1: Operation not permitted)
E: Sub-process /usr/bin/dpkg returned an error code (1)
FATAL:   post proc: exit status 100
FATAL:   While performing build: while running engine: exit status 255

### Steps to reproduce behavior
Singularity.def -
~~
BootStrap: docker
From: ubuntu:18.04

%post
  apt-get update
  apt-get -y install libsystemd0
~~

$ singularity build --fakeroot simg.sif Singularity.def

**Comment by cclerget** [comment]
Hi @Zohman , can you run this command :

```
singularity exec --fakeroot docker://alpine su - bin -s /bin/sh -c id
```

Additionally can you provide full build debug output with `singularity -d build --fakeroot simg.sif Singularity.def`

**Comment by Zohman** [comment]
Okay, i can't tell what changed.. but it seems to work now and i can't reproduce..

**Comment by cclerget** [comment]
Ok, closing the issue, feel free to re-open it if you encounter this issue again

**Comment by jscook2345** [comment]
@Zohman I think this is fixed via #4038 / #4047 I found a reproducer that worked
